crates/storage/provider/src/providers/static_file/manager.rs

```diff
@@ -898,14 +904,19 @@ impl StaticFileProvider {
                 self.get_segment_provider_from_transaction(segment, start, None)
             }
         };
-        let mut provider = get_provider(range.start)?;
 
+        let mut provider = Some(get_provider(range.start)?);
```

Comment:
I see we need to move the initial provider into the iterator and need to be able to drop it before requesting the next one